s2spy/rgdr/label_alignment.py

```diff
@@ -325,7 +356,8 @@ def _rename_datasets(
     """
     renamed = []
     for split, _rgdr in enumerate(rgdr_list):
-        data = clustered_data[split]
+        # A copy is required to not modify labels of the input data
+        data = copy(clustered_data[split])
```

Comment:
I wonder if the desired behaviour is to return copies. Before, it modified the data in place. The reason I commented on that is that the data was both updated in place, and returned. That made me think that you probably want one or the other, not both. Now, I'm wondering which one we actually want. Wouldn't it make more sense to update in place (and not return)?

Reply:
Seeing as xarray or pandas do not modify in place, I feel that returning a renamed copy is the most intuitive for the user. For example `xr.Dataset.rename`: "_Returns a new object with renamed variables, coordinates and dimensions_."

In notebooks, code modifying in-place can cause bugs that are hard to figure out when users don't run all cells in order (for example, when exploring things).

Reducing memory usage could be a reason for modifying in-place, however clustered data will not be memory intensive.

Reply:
Also, the calendar _does_ change in-place, but (at least to me) it feels closer to a Python `list`. While the label renaming feels more like an xarray/pandas operation.

Reply:
In that case, should the docstring of the `rename` function below, in line with the xarray example you mentioned, be something like "Returns a new object with renamed cluster labels."?

Reply:
I should probably improve the docstring to be more explicit then, yes.